lightning/src/chain/channelmonitor.rs

```diff
@@ -1870,17 +1835,43 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 		Vec::new()
 	}
 
-	/// Called by SimpleManyChannelMonitor::block_connected, which implements
-	/// ChainListener::block_connected.
-	/// Eventually this should be pub and, roughly, implement ChainListener, however this requires
-	/// &mut self, as well as returns new spendable outputs and outpoints to watch for spending of
-	/// on-chain.
-	fn block_connected<B: Deref, F: Deref, L: Deref>(&mut self, txn_matched: &[&Transaction], height: u32, block_hash: &BlockHash, broadcaster: B, fee_estimator: F, logger: L)-> Vec<(Txid, Vec<TxOut>)>
+	/// Checks if a given transaction matches the funding transaction or spends any watched outputs.
+	fn does_tx_match(&self, tx: &Transaction) -> bool {
+		let funding_txo = self.get_funding_txo();
+		if tx.txid() == funding_txo.0.txid {
+			for output in tx.output.iter() {
+				if output.script_pubkey == funding_txo.1 {
+					return true;
+				}
+			}
+		}
+
+		for input in tx.input.iter() {
+			for (txid, outputs) in self.get_outputs_to_watch().iter() {
+				for (idx, _script_pubkey) in outputs.iter().enumerate() {
+					if txid == &input.previous_output.txid && idx == input.previous_output.vout as usize {
+						return true;
+					}
+				}
+			}
+		}
+
+		false
+	}
+
+	/// Determines if any HTLCs have been resolved on chain in the connected block.
+	///
+	/// Returns any new outputs to watch from `txdata`. After called, these are also included in
```

Comment:
Hmm, I know its not your bug, but these are looking increasingly confusing without a struct that clearly defines what they mean (which I suppose was somewhat the case when it was a function to register watch outputs with documentation).